### PR TITLE
Apply RocketCDN CNAME change

### LINKED ISSUE #7027: RocketCDN CNAME change
**Is your feature request related to a problem? Please describe.**
In the upcoming version we'll need to change the RocketCDN CNAMEs automatically

**Describe the solution you'd like**
The Acceptance Criteria is described here:
https://www.notion.so/wpmedia/RocketCDN-CNAME-change-d72cd2baa88b49b7b4cb65b348440564?pvs=4

### PR DESCRIPTION
**Pull request by wordpressfan**
# Description

Fixes #7027

Starting from 3.17.3, we will show a message if the cname changed in WPR settings page only.

## Type of change

- [x] New feature (non-breaking change which adds functionality).

## Detailed scenario

the approach that we took here is that with update, we will force refreshing the rocketcdn status which has the cname url on it and compare it with the old/previous url then set an option which will be used to show the admin notice in WPR side, so the trigger here is the WPR plugin update.

### Documentation

With update we will force getting rocketcdn cname again and if it's different that the old url, we will show the notice to let the customers know about that.

### New dependencies

No

### Risks

I mentioned some questions with risks here: https://wp-media.slack.com/archives/C06CQPWEJSK/p1731572752314419
to @wp-media/product team

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

# Additional Checks
- [x] In the case of complex code, I wrote comments to explain it.
- [x] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [x] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

### CODE REVIEW

**Comment by wordpressfan** [comment]
I'll be waiting for a reply here: https://wp-media.slack.com/archives/C06CQPWEJSK/p1731572752314419
to see if we will change the direction that we took here or we are fine.

**Comment by wordpressfan** [comment]
I'll change the way we handle this to detect the old and new urls when setting rocketcdn_status transient instead of WPR upgrade.

File: inc/Engine/CDN/RocketCDN/NoticesSubscriber.php
Comment (remyperona) [inline inc/Engine/CDN/RocketCDN/NoticesSubscriber.php]: This would get the option from the `wp_rocket_settings` array, I don't think that's what you want

File: inc/Engine/WPRocketUninstall.php
Comment (remyperona) [inline inc/Engine/WPRocketUninstall.php]: the way you use the option, it's not a new one, but one that is part of the wp_rocket_settings array.

I think this should not be part of it, so you would need to use options_api only for managing the option, not Options_Data

File: inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php
Comment (remyperona) [inline inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php]: This probably should be options_api instead of options

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/Engine/WPRocketUninstall.php
Comment (wordpressfan) [reply inc/Engine/WPRocketUninstall.php]: That's a leftover from the old approach, removed

File: inc/Engine/CDN/RocketCDN/NoticesSubscriber.php
Comment (wordpressfan) [reply inc/Engine/CDN/RocketCDN/NoticesSubscriber.php]: That's what I wanted, I put this `rocketcdn_old_url` inside wp_rocket_settings main one.

File: inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php
Comment (wordpressfan) [reply inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php]: why?
I use `$this->options->set` to set the option inside our main settings option `wp_rocket_settings` then at the end I use `$this->options_api->set( 'settings', $this->options->get_options() );` to save all options, is this wrong?

Comment (remyperona) [reply inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php]: This should not be part of the `wp_rocket_settings` array, it's not an option of the plugin, and it would be removed the next time the settings are saved (because it's not part of the settings page form that is submitted)

Comment (wordpressfan) [reply inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php]: u r awesome @remyperona 
fixing that now

**Review by Khadreal — APPROVED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by remyperona** [comment]
PR approved to move to QA, but we should have tests coverage added to before merging

**Comment by piotrbak** [comment]
@wordpressfan There's still the question from Mathieu in this topic:
https://wp-media.slack.com/archives/CUT7FLHF1/p1731692989584199?thread_ts=1731601939.328049&cid=CUT7FLHF1

Please make after merging this PR to send it to transifex.

**Comment by Mai-Saad** [comment]
As per slack, 
Trigger: On update from < 3.17.3 to >= 3.17.3
If the CDN URL is set to xxxxxxxx.rocketcdn.me (where xxxxxxxx is exactly 8 alphanumeric characters), then:
change it to xxxxxxxx.delivery.rocketcdn.me
display the banner with before/after values
There should be no API call involved in this approach.
ACs would be: (validation is WIP)
- [x] On WP Rocket < 3.17.3, enable CDN option. Set xxxxxxxx.rocketcdn.me as CNAME URL in the plugin. Update to 3.17.3. The CNAME should be xxxxxxxx.delivery.rocketcdn.me and the banner must appear. 
- [x] On WP Rocket < 3.17.3, disable CDN option. Set xxxxxxxx.rocketcdn.me as CNAME URL in the plugin. Update to 3.17.3. The CNAME should be xxxxxxxx.delivery.rocketcdn.me and the banner must appear. 
- [x] On WP Rocket < 3.17.3, enable CDN option. Set xxxxxxxx.delivery.rocketcdn.me as CNAME URL in the plugin. Update to 3.17.3. The CNAME should still be xxxxxxxx.delivery.rocketcdn.me and the banner must not appear. => (from 3.16.4 and 3.17.2.1)
- [x] On WP Rocket < 3.17.3, enable CDN option. Set xxxxxxxx.another.cdn as CNAME URL in the plugin. Update to 3.17.3. The CNAME should still be xxxxxxxx.another.cdn and the banner must not appear. => (only green text under the option is displayed  on old/new version if we already have a valid subscription)


- [x] With Fresh install and activation of subscription => no banner displayed and CNAME with delivery is there :heavy_check_mark: 
- [x] Update from 3.17.3 to > 3.17.3 while xxxxxxxx.delivery.rocketcdn.me as CNAME => No banner displayed
- [x] Update from 3.17.3 to > 3.17.3 while  xxxxxxxx.rocketcdn.me as CNAME => ===> actual is no rewrite to cdn and no banner

**Comment by wordpressfan** [comment]
Slack discussion: https://wp-media.slack.com/archives/CUKB44GNN/p1731939168323999

We need to create another issue to add more automated tests (unit/integration) to fix the codacy CI failure.

**Pull request merged**
